content/articles/wavelet-transform-based-denoising-of-images-using-matlab/index.md

```diff
@@ -0,0 +1,141 @@
+Orthogonal wavelets are a single scaling function and the wavelet while biorthogonal generates a single scaling function and wavelet for 
+decomposition and another pair for reconstruction.
+Level decomposition involves passing signal through a low pass and high pass filter and the resultant signal is downsampled by 2. We obtain a 
+high-frequency signal with detailed co-efficient for high pass filter and low frequency for low pass filter with approximated co-efficient. 
+This is one level of decomposition(level one). When the process continues with the signal from the low pass filter, the level increases. The maximum 
+number of decomposition is calculated by;
+max. No. of decomposition = $$log_2$$N where N represents the signal length. The whole decomposition described above is the discrete wavelet transform. 
+The opposite of the above is the inverse discrete wavelet transform.
+
+Press the analyze pushbutton.
+When the analyzer pushbutton is pressed, our image is denoised. Out of the four buttons below the analyzer, we have the denoise button. Click on 
+this push button. This gives us a window with plots for the wavelet coefficient.
+
+![Plot of wavelet co-efficient](/engineering-education/wavelet-transform-based-denoising-of-images/image4.png)
+We have the detailed wavelet co-efficient. At level($$l_1$$), we the horizontal details, diagonal details, and vertical details. This also applies 
+to level 2($$l_2$$). On the left side of the window, select the threshold method and select the noise structure as well. In this case, we choose the 
+scaled white noise since it is Gaussian white noise. Using the scrollbar, we can vary the threshold value then click the denoise pushbutton.
+
+![Denoised image](/engineering-education/wavelet-transform-based-denoising-of-images/image5.png)
+We get the denoised image as shown above. This is how you can use the wavelet analyzer tool to the denoise image.
+
+### Functions used in Wavelet Denoising
+`wdencmp` - It can be used for denoising of both 1-D and 2-D signals. This means it can be used for both audios as well as images.
+`syntax`
+```Matlab
+imgden = wdencmp('gbl_or_Ivd',img, 'wname', N, THR, SORH, KEEPAPP)
+```
+where;
+img: is the noisy image that we input.
+imgden: is the output denoised image.
+gbl_or_ivd: either use `gbl` for single threshold or `Ivd` for level-dependent threshold(either use `gbl` or `Ivd`).
+wname: is the wavelet type used.
+N: represents the decomposition levels.
+THR: threshold levels[single value for `gbl` and 3xN matric to `ivd`, three rows each for horizontal, diagonal, and vertical detailed
+co-efficient and N number of columns, where N is the decomposition level].
+SORH: `s` or `h` corresponds to the soft and hard thresholding respectively.
+KEEPAPP: it is either 0 or 1. It is 0 when the approximation coefficients cannot be thresholded and 1 if they can be.
```

Comment:
Put these in bulleted points. And, use ` `` ` for highlighting the keywords/variables